Doc/library/sqlite3.rst

```diff
@@ -417,6 +417,18 @@ Module constants
    This flag may be combined with :const:`PARSE_COLNAMES` using the ``|``
    (bitwise or) operator.
 
+.. data:: SQLITE_OK
+          SQLITE_DENY
+          SQLITE_IGNORE
+
+   Flags used for return values from the callable passed as
+   the *authorizer_callback* argument of :meth:`Connection.set_authorizer`
+   to indicate whether:
+
+   * Access is allowed (:const:`!SQLITE_OK`),
+   * The SQL statement should be aborted with an error (:const:`!SQLITE_DENY`)
+   * The column should be treated as a ``NULL`` value (:const:`!SQLITE_IGNORE`)
```

Comment:
Unfortunately, it is not as straight-forward as this; for _SQLITE_READ actions_, this is indeed the case, but for _SQLITE_DELETE actions_, returning `SQLITE_IGNORE` apparently means that the truncate optimization is disabled, but the delete operation is still carried through (only row by row instead of the complete table at once). See https://www.sqlite.org/c3ref/set_authorizer.html

Perhaps we should build a table with the return values horizontally and the action codes vertically. That would imply also documenting the action codes, which will expand the PR significantly.

https://www.sqlite.org/c3ref/c_alter_table.html

Reply:
Unfortunately, my SQL/SQLite knowledge is mostly limited to what is actually in these docs, which is what these descriptions reflect (as I lifted them directly from the existing *authorizier_callback* reference). I'm not sure I'm really the person to be writing such (as opposed to the person reviewing it, from a beginner's perspective), so maybe it would be better to leave that to a followup PR by someone smarter than I (e.g. you), so long as this isn't a regression over the existing text.

Reply:
(TBH, I sometimes find the SQLite docs very hard to read.)

Reply:
Well if they are hard to read for you, then I'm sure they are likely downright impossible for me (though to be honest, the few specific bits I've read haven't been the worst)